File: examples/docs-code-snippets/src/main/scala/examples/scala/interactions/CancelOrder.scala
package examples.scala.interactions

import com.ing.baker.recipe.scaladsl.{Event, Ingredient, Interaction}
import examples.scala.ingredients.Address

object CancelOrder {

  case class OrderCancelled()

  val interaction: Interaction = Interaction(
    name = "CancelOrder",
    inputIngredients = Seq(
      Ingredient[String](name = "orderId"),
      Ingredient[List[String]](name = "unavailableProductIds")
    ),
    output = Seq(
      Event[OrderCancelled]
    )
  )
}

File: examples/docs-code-snippets/src/main/java/examples/java/interactions/CancelOrder.java
package examples.java.interactions;

import com.ing.baker.recipe.annotations.FiresEvent;
import com.ing.baker.recipe.annotations.RequiresIngredient;
import com.ing.baker.recipe.javadsl.Interaction;

import java.util.List;

public interface CancelOrder extends Interaction {

    record OrderCancelled() {
    }

    @FiresEvent(oneOf = {OrderCancelled.class})
    OrderCancelled apply(@RequiresIngredient("orderId") String orderId,
                         @RequiresIngredient("unavailableProductIds") List<String> unavailableProductIds
    );
}
